src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -789,19 +789,15 @@ public override bool Equals(object? o)
             else g = (Guid)o;
 
             // Now compare each of the elements
-            return g._a == _a &&
-                Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
-                Unsafe.Add(ref g._a, 2) == Unsafe.Add(ref _a, 2) &&
-                Unsafe.Add(ref g._a, 3) == Unsafe.Add(ref _a, 3);
+            return Unsafe.As<int, long>(ref g._a) == Unsafe.As<int, long>(ref _a) &&
+                Unsafe.As<byte, long>(ref g._d) == Unsafe.As<byte, long>(ref _d);
         }
 
         public bool Equals(Guid g)
         {
             // Now compare each of the elements
-            return g._a == _a &&
-                Unsafe.Add(ref g._a, 1) == Unsafe.Add(ref _a, 1) &&
-                Unsafe.Add(ref g._a, 2) == Unsafe.Add(ref _a, 2) &&
-                Unsafe.Add(ref g._a, 3) == Unsafe.Add(ref _a, 3);
+            return Unsafe.As<int, long>(ref g._a) == Unsafe.As<int, long>(ref _a) &&
+                Unsafe.As<byte, long>(ref g._d) == Unsafe.As<byte, long>(ref _d);
```

Comment:
I was under the impression that Int64-based operations would be slower than Int32-based operations on 32-bit, especially if the values weren't aligned (and that that's why @jkotas used Int32 here initially when switching these operations to be based on Unsafe).  Is that not the case?